src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -111,6 +127,54 @@ internal static int GetMemberArityIncludingExtension(this Symbol member)
             return member.GetMemberArity();
         }
 
+        internal static ImmutableArray<TypeParameterSymbol> GetTypeParametersIncludingExtension<TMember>(this TMember member) where TMember : Symbol
+        {
+            if (member is MethodSymbol method)
+            {
+                return method.GetIsNewExtensionMember()
+                    ? method.ContainingType.TypeParameters.Concat(method.TypeParameters)
+                    : method.ConstructedFrom.TypeParameters;
+            }
+
+            if (member is PropertySymbol property)
+            {
+                Debug.Assert(property.GetIsNewExtensionMember());
+                return property.ContainingType.TypeParameters;
+            }
+
+            throw ExceptionUtilities.UnexpectedValue(member);
+        }
+
+        internal static PooledDictionary<TypeParameterSymbol, int>? MakeOrdinalsIfNeeded<TMember>(this TMember member, ImmutableArray<TypeParameterSymbol> originalTypeParameters)
+            where TMember : Symbol
+        {
+            if (member is MethodSymbol method)
+            {
+                PooledDictionary<TypeParameterSymbol, int>? ordinals = null;
+                if (method.GetIsNewExtensionMember() && method.Arity > 0 && method.ContainingType.Arity > 0)
+                {
+                    Debug.Assert(originalTypeParameters.Length == method.Arity + method.ContainingType.Arity);
+
+                    // Since we're concatenating type parameters from the extension and from the method together
+                    // we need to control the ordinals that are used
+                    ordinals = PooledDictionary<TypeParameterSymbol, int>.GetInstance();
```

Comment:
>ordinals = PooledDictionary<TypeParameterSymbol, int>.GetInstance(); [](http://example.com/codeflow?start=20&length=68)

I think the map that we create here is supposed to use identity comparison. Is this what we get?

Reply:
This thread was resolved, but I do not think changing the key to `object` changes anything about how keys are compared